From f3db3fd2c8adcf6655c24ac20034c2a4fc068280 Mon Sep 17 00:00:00 2001
From: Akira TAGOH <akira@tagoh.org>
Date: Mon, 2 Oct 2017 21:17:06 +0900
Subject: [PATCH 6/7] Don't call FcStat when the alias has already been added

We could assume that the targeted location is mapped at the different place
when there are in the alias table.
---
 src/fccfg.c | 21 +++++++--------------
 1 file changed, 7 insertions(+), 14 deletions(-)

diff --git a/src/fccfg.c b/src/fccfg.c
index 514357e..29acd1d 100644
--- a/src/fccfg.c
+++ b/src/fccfg.c
@@ -349,24 +349,17 @@ FcConfigAddCache (FcConfig *config, FcCache *cache,
 	for (i = 0; i < cache->dirs_count; i++)
 	{
 	    const FcChar8 *dir = FcCacheSubdir (cache, i);
+	    const FcChar8 *alias;
+	    FcChar8 *d = FcStrDirname (dir);
 	    FcChar8 *s = NULL;
-	    struct stat statb;
 
-	    if (FcStat (dir, &statb) < 0)
+	    if ((alias = FcDirCacheFindAliasPath (d)))
 	    {
-		const FcChar8 *alias;
-		FcChar8 *d = FcStrDirname (dir);
-
-		if ((alias = FcDirCacheFindAliasPath (dir)))
-		    dir = alias;
-		else if ((alias = FcDirCacheFindAliasPath (d)))
-		{
-		    FcChar8 *base = FcStrBasename (dir);
-		    dir = s = FcStrBuildFilename (alias, base, NULL);
-		    FcStrFree (base);
-		}
-		FcStrFree (d);
+		FcChar8 *base = FcStrBasename (dir);
+		dir = s = FcStrBuildFilename (alias, base, NULL);
+		FcStrFree (base);
 	    }
+	    FcStrFree (d);
 	    if (FcConfigAcceptFilename (config, dir))
 		FcStrSetAddFilename (dirSet, dir);
 	    if (s)
-- 
2.13.3

